keanu-project/src/test/java/io/improbable/keanu/e2e/regression/RadonHeirarchicalRegression.java

```diff
@@ -0,0 +1,179 @@
+
+public class RadonHeirarchicalRegression {
+
+    @Rule
+    public DeterministicRule rule = new DeterministicRule();
+
+    private List<Data> radonData;
+
+    @Before
+    public void readRadonCSV() {
+        radonData = ReadCsv.fromResources("data/datasets/radon/radon.csv")
+            .asRowsDefinedBy(Data.class)
+            .load(true);
+    }
+
+    @Test
+    public void canPerformSimpleLinearRegression() {
+        RegressionModel model = linearRegression(radonData);
+        Assert.assertTrue(model.getWeight(0) > -0.7 && model.getWeight(0) < -0.4);
+        Assert.assertTrue(model.getIntercept() > 1.2 && model.getIntercept() < 1.5);
+    }
+
+    @Test
+    public void canPerformRegressionWithOneHeirarchy() {
+        buildHeirarchicalNetwork(radonData, 1);
+    }
+
+    @Test
+    public void canPerformRegressionWithTwoHeirarchies() {
+        buildHeirarchicalNetwork(radonData, 2);
+    }
+
+    @Test
+    public void canPerformRegressionWithFourHeirarchies() {
+        buildHeirarchicalNetwork(radonData, 4);
+    }
+
+    @Test
+    public void canPerformRegressionWithTenHeirarchies() {
+        buildHeirarchicalNetwork(radonData, 10);
+    }
+
+    private RegressionModel linearRegression(List<Data> data) {
+        double[] radon = data.stream().mapToDouble(k -> k.log_radon).toArray();
+        double[] floor = data.stream().mapToDouble(k -> k.floor).toArray();
+        DoubleTensor y = DoubleTensor.create(radon, 1, radon.length);
+        DoubleTensor x = DoubleTensor.create(floor, 1, floor.length);
+
+        RegressionModel model = RegressionModel.
+            withTrainingData(x, y).
+            withRegularization(RegressionRegularization.RIDGE).
+            withPriorOnWeights(0., 5.).
+            withPriorOnIntercept(0., 5.).
+            build();
+
+        model.observe();
+        model.fit();
+
+        return model;
+    }
+
+    private void buildHeirarchicalNetwork(List<Data> radonData, int numberOfModels) {
+        GaussianVertex muAlpha = new GaussianVertex(new long[]{1, 1}, 0, 100).setLabel("MuIntercept");
+        GaussianVertex muBeta = new GaussianVertex(new long[]{1, 1}, 0, 100).setLabel("MuGradient");
+
+        HalfGaussianVertex sigmaAlpha = new HalfGaussianVertex(new long[]{1, 1}, 10.).setLabel("SigmaIntercept");
+        HalfGaussianVertex sigmaBeta = new HalfGaussianVertex(new long[]{1, 1}, 10.).setLabel("SigmaGradient");
+
+        int numPartitions = radonData.size() / numberOfModels;
+
+        double[] allFloor = radonData.stream().mapToDouble(k -> k.floor).toArray();
+        double[] allRadon = radonData.stream().mapToDouble(k -> k.log_radon).toArray();
+
+        List<RegressionModel> models = new ArrayList<>();
+
+        for (int i = 0; i < numberOfModels; i++) {
+            RegressionModel model = createSubModel(allFloor, allRadon, i, numPartitions, muBeta, muAlpha, sigmaBeta, sigmaAlpha);
+            models.add(model);
+        }
+
+        muAlpha.setValue(1.);
```

Comment:
A comment to why these value are set would be useful as it's a bit of an odd case.